### ISSUE TITLE
Cannot find name 'ProgressEvent' when compiling TypeScript

### ISSUE THREAD

**Issue by claust** [axios/axios#3219]
#### Describe the bug
When trying to compile a Firebase function (Node.js / TypeScript) I get an error:
```js
node_modules/axios/index.d.ts:62:38 - error TS2304: Cannot find name 'ProgressEvent'.

62   onUploadProgress?: (progressEvent: ProgressEvent) => void;
                                        ~~~~~~~~~~~~~

node_modules/axios/index.d.ts:63:40 - error TS2304: Cannot find name 'ProgressEvent'.

63   onDownloadProgress?: (progressEvent: ProgressEvent) => void;
                                          ~~~~~~~~~~~~~
```

#### To Reproduce
Example of a `package.json` that fails to compile using `npm run build`.
```js
{
  "name": "functions",
  "scripts": {
    "lint": "eslint -c .eslintrc.js --ext .ts ./src",
    "lint-fix": "eslint -c .eslintrc.js --fix --ext .ts ./src",
    "build": "tsc",
    "serve": "npm run build && firebase serve --only functions",
    "shell": "npm run build && firebase functions:shell",
    "start": "npm run shell",
    "firebase": "firebase serve --only functions --port=8080",
    "deploy": "npm run build && npm run lint && firebase deploy --only functions",
    "logs": "firebase functions:log"
  },
  "main": "lib/index.js",
  "dependencies": {
    "axios": "^0.20.0",
    "firebase-functions": "^3.10.0"
  },
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^3.9.1",
    "@typescript-eslint/parser": "^3.9.1",
    "eslint": "^7.7.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-prettier": "^3.1.4",
    "prettier": "^2.0.5",
    "typescript": "^3.9.7"
  },
  "private": true,
  "engines": {
    "node": "10"
  }
}
```

#### Expected behavior
The Node.js code should compile.

#### Environment
 - Axios Version: 0.20.0
 - Node.js Version: 10.21.0
 - OS: Ubuntu 20.04

#### Additional context/Screenshots
None.

**Comment by jan-molak** [comment]
Hey folks, I'd like to suggest to avoid making Axios depend on lib dom by using triple-slash directives and use a structural interface instead of `ProgressEvent`. 

Please see my proposal at #3228.

Thanks,
Jan

**Comment by Mister-Hope** [comment]
Any progress on this one? It's a critical bug which can not be bypassed if typescript is running on pure node js env.

**Comment by FlorianPfisterer** [comment]
@Mister-Hope In case you don't need any 0.20.0 features, check out [my comment](https://github.com/axios/axios/issues/3219#issuecomment-678332521) above for a fix.

**Comment by Mister-Hope** [comment]
> @Mister-Hope In case you don't need any 0.20.0 features, check out [my comment](https://github.com/axios/axios/issues/3219#issuecomment-678332521) above for a fix.

Just a little curious that nearly half a month has passed, and it's not fixed yet.

**Comment by Kryshtopenko** [comment]
nice ...

**Comment by Decimator** [comment]
Are there plans to merge @jan-molak's proposed PR? Thanks for the work on this

**Comment by Mister-Hope** [comment]
> Are there plans to merge @jan-molak's proposed PR? Thanks for the work on this.
> 
> _edit_: adding "DOM" to tsconfig "lib" array is a good workaround btw, tested and it appears to work!

Not with me, including dom will trigger some issues in my project.

**Comment by ruslanguns** [comment]
I temporary fixed by using the resolution.

```json
// package.json
{
"resolutions": {
    "axios": "^0.19.0"
  }
}

```

**Comment by ksugiarto** [comment]
Downgrading is not working on my case, since I use Nest, and Nest latest has its dependency to axios@20.0.0

```
"@nestjs/common": {
  "version": "7.4.4",
  "resolved": "https://registry.npmjs.org/@nestjs/common/-/common-7.4.4.tgz",
  "integrity": "sha512-Cj94FJrnLcAU4URJrRmnsHRODZPJpX+EKKJ/Or9qvL9ULQwRWYmFcGQYaJ0nVV0hSBjn/jaAV1Cgqw74uk21KA==",
  "requires": {
    "axios": "0.20.0",
    "cli-color": "2.0.0",
    "iterare": "1.2.1",
    "tslib": "2.0.1",
    "uuid": "8.3.0"
  },
  "dependencies": {
    "axios": {
      "version": "0.20.0",
      "resolved": "https://registry.npmjs.org/axios/-/axios-0.20.0.tgz",
      "integrity": "sha512-ANA4rr2BDcmmAQLOKft2fufrtuvlqR+cXNNinUmvfeSNCOF98PZL+7M/v1zIdGo7OLjEA9J2gXJL+j4zGsl0bA==",
      "requires": {
        "follow-redirects": "^1.10.0"
      }
    }
  }
},
```

So even I downgrade Axios alone, it won't fix the error on my case, I need to downgrade most of my nest core lib to do so. Hm, is that any other workaround for now?.

**Comment by Mister-Hope** [comment]
> Just a little curious that nearly half a month has passed, and it's not fixed yet.

Now nearly a month passed, and I saw there is already a PR #3228

**Comment by sajithneyo** [comment]
Still exists in Axios 0.20.0

**Comment by AuspeXeu** [comment]
@sajithneyo it'll be in the 0.20.1 release I think?! @jasonsaayman

**Comment by ahippler** [comment]
@emilyemorehouse when will be the release with this fix? This bug is open for more 2 month...

**Comment by Mister-Hope** [comment]
Don't know, have to say that typescript users like me are cathing bad experience with it in the past 2 month,

**Comment by Mister-Hope** [comment]
I don't want to hurt anyone, but leaving the bugs for month in a package with
![image](https://user-images.githubusercontent.com/33315834/95987026-743d8880-0e59-11eb-90c0-a1b4b61b0a5d.png)
is not a good idea, I believe there are plenty of typescript users are suffering this problem on pure node env, and have to downgrade back to 0.20.0 while editing their dependency bots(or ci) to omit axios.

**Comment by Mister-Hope** [comment]
It's fixed so it can be closed, but it's.not released.

**Comment by jasonsaayman** [comment]
HI, It will hopefully be released today or over the weekend 😄

**Comment by AuspeXeu** [comment]
@jasonsaayman just out of curiosity, who's authorized to publish a new release?

**Comment by jasonsaayman** [comment]
Hi @AuspeXeu,

At the moment only Emily can do this, Nick and Matt could but they are not actively maintaining the project as they handed over to Emily. I know Emily is pretty busy with stuff but I do have direct communication with her and the release is planned to go out in the next few days.

**Comment by AuspeXeu** [comment]
Ah, sweet. Given how many people rely on axios, it might be good if you also have this sort of authorization. Maybe worth mentioning it to her :)

**Comment by g14a** [comment]
Adding "DOM" to the tsconfig lib field works somehow :)

**Comment by tomcanac** [comment]
Hi,

Thanks for the great library and the work, it's really appreciated to be able to have this library up and running 🎉!
We are hitting this bug since our last package update, do you have any news about the publishing of the latest fix?

Thanks, really appreciated!

**Comment by jasonsaayman** [comment]
Just to update everyone I have asked again today to see when the release will happen, will update this thread as soon as I know whats going on.

**Comment by Mister-Hope** [comment]
> > Please reopen, installed axios today and issue is still present!
> 
> It's fixed so it can be closed, but it's.not released.

@AuspeXeu Why are you sending me the confused emoji? An issue should be closed the time a linked PR is merged.

**Comment by emmanuelagarry** [comment]
> Adding "DOM" to the tsconfig lib field works somehow :)

This worked like a charm

**Comment by jasonsaayman** [comment]
Hi,

Version 0.21.0 has been released 🎉 please use that and let us know if that solves your issue.

Thanks

**Comment by alexpls** [comment]
> Version 0.21.0 has been released 🎉 please use that and let us know if that solves your issue.

Upgraded to v0.21.0 and can confirm that we can now build successfully without needing to include the `dom` library. Thank you!

**Comment by pSnehanshu** [comment]
I am getting this error as of today

```
node_modules/axios/index.d.ts:275:11 - error TS2304: Cannot find name 'ProgressEvent'.

275   event?: ProgressEvent;
              ~~~~~~~~~~~~~


Found 1 error in node_modules/axios/index.d.ts:275
```

* Axios version: 1.2.0
* Node version: 18.12.1
* tsconfig.json:
```json
{
  "compilerOptions": {
    "sourceMap": true,
    "outDir": "dist",
    "strict": true,
    "lib": ["esnext"],
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "target": "ES6",
    "module": "CommonJS"
  },
  "include": ["src"],
  "ts-node": {
    "files": true
  }
}
```

**Comment by thinktapper** [comment]
I also had this error today & am happy to share that the fix has been merged as of [v1.2.1](https://github.com/axios/axios/releases/tag/v1.2.1). Updating fixed it for me.
